src/util_test.cc

```diff
@@ -428,3 +428,13 @@ TEST(ElideMiddle, ElideInTheMiddle) {
   EXPECT_EQ("012...789", elided);
   EXPECT_EQ("01234567...23456789", ElideMiddle(input, 19));
 }
+
+TEST(ElideMiddle, NothingToElideMiddleWithEscape) {
+  string input = "Nothing \33[m string.";
+  EXPECT_EQ("Nothing \33[m string.", ElideMiddle(input, 80));
```

Comment:
80 is enough for an non-ANSI-aware `ElideMiddle`. Please check that it correctly starts to elide at the right time ans move this nothing-to-elide check into the `ElideMiddleWidthEscape` test.